sqlalchemy-stubs/util/_collections.pyi

```diff
@@ -1,101 +1,142 @@
-    def __delitem__(self, key): ...
-    def pop(self, key, *default): ...
-    def popitem(self): ...
-
-class OrderedSet(set):
-    def __init__(self, d: Optional[Any] = ...) -> None: ...
-    def add(self, element): ...
-    def remove(self, element): ...
-    def insert(self, pos, element): ...
-    def discard(self, element): ...
-    def clear(self): ...
-    def __getitem__(self, key): ...
-    def __iter__(self): ...
-    def __add__(self, other): ...
-    def update(self, iterable): ...
-    __ior__: Any = ...
-    def union(self, other): ...
-    __or__: Any = ...
-    def intersection(self, other): ...
-    __and__: Any = ...
-    def symmetric_difference(self, other): ...
-    __xor__: Any = ...
-    def difference(self, other): ...
-    __sub__: Any = ...
-    def intersection_update(self, other): ...
-    __iand__: Any = ...
-    def symmetric_difference_update(self, other): ...
-    __ixor__: Any = ...
-    def difference_update(self, other): ...
-    __isub__: Any = ...
+    def clear(self) -> None: ...
+    def copy(self) -> OrderedDict[_KT, _VT]: ...
+    def __copy__(self) -> OrderedDict[_KT, _VT]: ...
+    if sys.version_info < (3, 0):
+        def sort(self, cmp: Callable[[_VT, _VT], Any] = ..., key: Callable[[_VT], Any] = ..., reverse: bool = ...) -> None: ...
+    else:
+        def sort(self, *, key: Optional[Callable[[_VT], Any]] = ..., reverse: bool = ...) -> None: ...
+    @overload
+    def update(self, ____sequence: Mapping[_KT, _VT], **kwargs: _VT) -> None: ...
+    @overload
+    def update(self, ____sequence: Iterable[Tuple[_KT, _VT]], **kwargs: _VT) -> None: ...
+    @overload
+    def update(self, **kwargs: _VT) -> None: ...
+    # Signature of "setdefault" incompatible with supertype "dict" and "MutableMapping"
+    def setdefault(self, key: _KT, value: _VT) -> _VT: ...  # type: ignore
+    def __iter__(self) -> Iterator[_KT]: ...
+    # Return type of "keys" incompatible with supertype "dict" and "Mapping"
+    def keys(self) -> List[_KT]: ...  # type: ignore
+    # Return type of "values" incompatible with supertype "dict" and "Mapping"
+    def values(self) -> List[_VT]: ...  # type: ignore
+    # Return type of "items" incompatible with supertype "dict" and "Mapping"
+    def items(self) -> List[Tuple[_KT, _VT]]: ...  # type: ignore
+    if sys.version_info < (3, 0):
+        def itervalues(self) -> Iterator[_VT]: ...
+        def iterkeys(self) -> Iterator[_KT]: ...
+        def iteritems(self) -> Iterator[Tuple[_KT, _VT]]: ...
+    def __setitem__(self, key: _KT, object: _VT) -> None: ...
+    def __delitem__(self, key: _KT) -> None: ...
+    @overload
+    def pop(self, key: _KT) -> _VT: ...
+    @overload
+    def pop(self, key: _KT, *default: Union[_VT, _T]) -> Union[_VT, _T]: ...
+    def popitem(self) -> Tuple[_KT, _VT]: ...
+
+_OS = TypeVar('_OS', bound=OrderedSet)
+
+class OrderedSet(Set[_T]):
+    def __init__(self, d: Optional[Iterable[_T]] = ...) -> None: ...
```

Comment:
I would add a TODO item here.
```suggestion
    # TODO: double-check method signatures in this class.
    # We can potentially find some compromises to avoid a bunch of '# type: ignore's
    def __init__(self, d: Optional[Iterable[_T]] = ...) -> None: ...
```